API_GUIDE.md

```diff
@@ -143,11 +154,23 @@ PyTorch/XLA behaves semantically like regular PyTorch and XLA tensors, implement
     ```
 
 
-3.  Loops with a different number of iterations between steps are subject to similar observations as tensor shapes. PyTorch/XLA automatically handles them, but they are seen as different execution graphs and require recompilations.
+1.  Loops with a different number of iterations between steps are subject to similar observations as tensor shapes. PyTorch/XLA automatically handles them, but they are seen as different execution graphs and require recompilations. Therefore, to get the best performance, same computations should run on all XLA devices in all hosts.
+One example of this is: iterators in `torch_xla_py.data_parallel` may drop the
```

Comment:
This should not be an example of that. Should have it own bullet point, and mention that with very small sample counts, it is better to use small batch sizes.